tools/make-bots

```diff
@@ -3,14 +3,16 @@
 # Prepare bots by creating ./bots directory
 # Specify $COCKPIT_BOTS_REF to checkout non-master branch
 
-set -e
+GITHUB_REPO='bots'
+CLONE_TARGET='bots'
+
+set -eu
+cd "$(realpath -m "$0"/../..)"
+. tools/git-utils.sh
 
 if [ ! -d bots ]; then
-    git clone --quiet --reference-if-able ${XDG_CACHE_HOME:-$HOME/.cache}/cockpit-project/bots https://github.com/cockpit-project/bots.git
-    if [ -n "$COCKPIT_BOTS_REF" ]; then
-        git -C bots fetch --quiet --depth=1 origin $COCKPIT_BOTS_REF
-        git -C bots checkout --quiet FETCH_HEAD
-    fi
+    fetch_to_cache # it's small, so keep everything cached
+    clone_from_cache "${COCKPIT_BOTS_REF-master}"
```

Comment:
https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html § 2.6.2:

> `${parameter:-[word]}`
>    Use Default Values. If parameter is unset or null, the expansion of word (or an empty string if word is omitted) shall be substituted; otherwise, the value of parameter shall be substituted.

...and just below...

> In the parameter expansions shown previously, use of the `<colon>` in the format shall result in a test for a parameter that is unset or null; omission of the `<colon>` shall result in a test for a parameter that is only unset.